invokeai/backend/model_management/model_manager.py

```diff
@@ -982,9 +1054,9 @@ def preamble(self) -> str:
     def _disambiguate_name(self, model_name:str, model_type:SDModelType)->str:
         model_type = model_type or SDModelType.diffusers
         full_name = f"{model_name}/{model_type.name}"
-        if self.valid_model(full_name):
+        if full_name in self.config:
             return full_name
-        if self.valid_model(model_name):
+        if model_name in self.config:
```

Comment:
Didn't it falsely triggered when you have for example lora and diffusers with same name?
So, you call `valid_model(lora_name, Type.Lora)` which calls this function and if you have something else with same name as `lora_name` then it will return it like this is lora.
IMHO - we should always use key for config with type and name. It's easier on download to concat type with name then tries to handle different formats of key.

Reply:
I want to be able to handle legacy `models.yaml`. I have just committed a fix that allows the bare name to be used if and only if the type requested is 'diffusers' and the stanza's `format` is `diffusers` as well.

The model manager `import` methods all write out stanzas using the `name/type` syntax, so new `models.yaml` will be correct.